### ISSUE TITLE
False-positive `InvalidReturnType` on `getOptions()` when option shortcut is null

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1165]
## Summary

Psalm reports a false-positive `InvalidReturnType` / `InvalidReturnStatement` on any console command that overrides `getOptions()` (or `getArguments()`) and passes `null` as the option **shortcut**, which is the most common case (no shortcut).

## Reproduction

```php
<?php declare(strict_types=1);

use Symfony\Component\Console\Input\InputOption;

final class ProbeCommand extends \Illuminate\Console\Command
{
    /** @inheritDoc */
    #[\Override]
    protected function getOptions(): array
    {
        return [
            ['nullShortcut', null, InputOption::VALUE_REQUIRED, 'desc', 'default'],
        ];
    }
}
```

Psalm output:

```
InvalidReturnType: The declared return type
'array<array-key, InputOption|array{0: non-empty-string,
  1?: non-empty-array<array-key, string>|string, 2?: 0|1|...|31,
  3?: string, 4?: mixed, 5?: ...}>'
for getOptions is incorrect, got
'list{list{'nullShortcut', null, 2, 'desc', 'default'}}'
```

Using a non-null shortcut (`'n'` instead of `null`) makes the error disappear, which isolates the cause to element index `1` (the shortcut).

## Root cause

The return type comes from `laravel/framework`'s native docblock in
`Illuminate\Console\Concerns\HasParameters::getOptions()` (verified on `v12.62.0` and current `master`):

```php
/**
 * @return (InputOption|array{
 *    0: non-empty-string,
 *    1?: string|non-empty-array<string>,   // <-- missing |null
 *    2?: int-mask-of<InputOption::VALUE_*>,
 *    3?: string,
 *    4?: mixed,
 *    5?: list<string|Suggestion>|\Closure(CompletionInput, CompletionSuggestions): list<string|Suggestion>
 * })[]
 */
protected function getOptions()
```

But Symfony's `InputOption` constructor accepts a nullable shortcut, defaulting to `null`:

```php
public function __construct(
    string $name,
    string|array|null $shortcut = null,
    ...
)
```

So index `1` should be `string|non-empty-array<string>|null`. A null shortcut (no shortcut) is valid and is the typical case.

## Suggested fix

The cleanest fix is upstream in `laravel/framework` (`HasParameters::getOptions()`, add `|null` at index 1). If a framework PR isn't desired, the plugin could ship a stub override for `HasParameters::getOptions()` / `getArguments()` with the corrected shortcut type so users aren't forced to baseline this.

## Versions

- `psalm/plugin-laravel`: dev-master
- `laravel/framework`: v12.62.0 (also reproduces on master)
- `symfony/console`: shortcut signature `string|array|null $shortcut = null`
